app/scripts/controllers/network/createInfuraClient.test.js

```diff
@@ -0,0 +1,324 @@
+        numberOfParameters: 1,
+      });
+    });
+
+    describe('eth_newPendingTransactionFilter', () => {
+      testsForRpcMethodNotHandledByMiddleware(
+        'eth_newPendingTransactionFilter',
+        { numberOfParameters: 0 },
+      );
+    });
+
+    describe('eth_protocolVersion', () => {
+      testsForRpcMethodAssumingNoBlockParam('eth_protocolVersion');
+    });
+
+    describe('eth_sendRawTransaction', () => {
+      testsForRpcMethodNotHandledByMiddleware('eth_sendRawTransaction', {
+        numberOfParameters: 1,
+      });
+    });
+
+    describe('eth_sendTransaction', () => {
+      testsForRpcMethodNotHandledByMiddleware('eth_sendTransaction', {
+        numberOfParameters: 1,
+      });
+    });
+
+    describe('eth_sign', () => {
+      testsForRpcMethodNotHandledByMiddleware('eth_sign', {
+        numberOfParameters: 2,
+      });
+    });
+
+    describe('eth_submitWork', () => {
+      testsForRpcMethodNotHandledByMiddleware('eth_submitWork', {
+        numberOfParameters: 3,
+      });
+    });
+
+    describe('eth_syncing', () => {
+      testsForRpcMethodNotHandledByMiddleware('eth_syncing', {
+        numberOfParameters: 0,
+      });
+    });
+
+    describe('eth_uninstallFilter', () => {
+      testsForRpcMethodNotHandledByMiddleware('eth_uninstallFilter', {
+        numberOfParameters: 1,
+      });
+    });
+  });
+
+  describe('RPC methods supported by Infura but not listed in the JSON-RPC spec', () => {
+    describe('eth_subscribe', () => {
+      testsForRpcMethodNotHandledByMiddleware('eth_subscribe', {
+        numberOfParameters: 1,
+      });
+    });
+
+    describe('eth_unsubscribe', () => {
+      testsForRpcMethodNotHandledByMiddleware('eth_unsubscribe', {
+        numberOfParameters: 1,
+      });
+    });
+
+    describe('net_listening', () => {
+      testsForRpcMethodNotHandledByMiddleware('net_listening', {
+        numberOfParameters: 0,
+      });
+    });
+
+    describe('net_peerCount', () => {
+      testsForRpcMethodNotHandledByMiddleware('net_peerCount', {
+        numberOfParameters: 0,
+      });
+    });
+
+    describe('net_version', () => {
+      it('does not hit Infura, instead returning the chain id that maps to the Infura network, as a decimal string', async () => {
+        const chainId = await withInfuraClient(
+          { network: 'ropsten' },
+          ({ makeRpcCall }) => {
+            return makeRpcCall({
+              method: 'net_version',
+            });
+          },
+        );
+
+        expect(chainId).toStrictEqual('3');
+      });
+    });
+
+    describe('parity_nextNonce', () => {
+      testsForRpcMethodNotHandledByMiddleware('parity_nextNonce', {
+        numberOfParameters: 1,
+      });
+    });
+
+    describe('web3_clientVersion', () => {
+      testsForRpcMethodAssumingNoBlockParam('web3_clientVersion');
+    });
+  });
+
+  // NOTE: The following methods are omitted because although they are listed in
```

Comment:
Nit: I don't think it would hurt to include tests for these methods as well.

I didn't directly respond to your "What should be the source of truth" question before because I didn't really understand what you meant. I don't think we should consider either resource as a "source of truth" on what we should test, but rather a combination of factors. I think it's useful to test anything we have special behaviour for (e.g. the methods cached by the middleware), and any we anticipate adding special behaviour for in the future (Ethereum standard methods, and maybe more eventually). But this doesn't need to be a blocker.

Reply:
Per our conversation we decided to handle this in another PR.